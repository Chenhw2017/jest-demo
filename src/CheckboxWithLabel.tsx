import React from 'react';

interface ILinkProps {
  labelOn: string,
  labelOff: string
}

interface ILinkState {
  isChecked: boolean
}

export default class CheckboxWithLabel extends React.Component <ILinkProps, ILinkState> {
  state = {
    isChecked: false,
  };

  onChange = () => {
    this.setState({isChecked: !this.state.isChecked});
  };

  render() {
    return (
      <label>
        <input
          data="3423423"
          type="checkbox"
          checked={this.state.isChecked}
          onChange={this.onChange}
        />
        {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
      </label>
    );
  }
}