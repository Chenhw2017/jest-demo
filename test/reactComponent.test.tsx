import React from 'react';
import Enzyme, { shallow, render } from 'enzyme';
import renderer from 'react-test-renderer';
// import reactDom from 'react-dom';
import glob from "glob";

import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import CheckboxWithLabel from '../src/CheckboxWithLabel';

describe('learn Enzyme', () => {
  // console.log(glob.sync('./src/*.tsx'))
  it('CheckboxWithLabel changes the text after click', () => {
    // Render a checkbox with label in the document
    const checkbox = shallow(<CheckboxWithLabel labelOn="On" labelOff="Off" />);
    const wrapper = render(<CheckboxWithLabel labelOn="On" labelOff="Off" />);
    expect(wrapper).toMatchSnapshot();
    expect(checkbox.text()).toEqual('Off');

    checkbox.find('input').simulate('change');

    expect(checkbox.text()).toEqual('On');
  });


  it('react-test-renderer CheckboxWithLabel changes the text after click', () => {
    const tree = renderer
      .create(<CheckboxWithLabel labelOn="On" labelOff="Off" />)
      .toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
})