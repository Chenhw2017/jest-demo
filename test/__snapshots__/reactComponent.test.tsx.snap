// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`learn Enzyme CheckboxWithLabel changes the text after click 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "data": "3423423",
          "type": "checkbox",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": "Off",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "data": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "data": undefined,
          "type": undefined,
        },
      },
      Object {
        "data": "Off",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "data": "3423423",
            "type": "checkbox",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "data": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "data": undefined,
            "type": undefined,
          },
        },
        "type": "text",
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`learn Enzyme react-test-renderer CheckboxWithLabel changes the text after click 1`] = `
<label>
  <input
    checked={false}
    data="3423423"
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;
